keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -84,10 +85,11 @@ public static Nd4jDoubleTensor arange(double start, double end, double stepSize)
     private INDArray tensor;
 
     public Nd4jDoubleTensor(double[] data, int[] shape) {
-        this.tensor = TypedINDArrayFactory.create(data, shape, BUFFER_TYPE);
+        this(TypedINDArrayFactory.create(data, shape, BUFFER_TYPE));
     }
 
     public Nd4jDoubleTensor(INDArray tensor) {
+        TensorShapeValidation.checkRankIsAtLeastTwo(tensor.shape());
```

Comment:
It's great that we're doing this now, takes the human out of making sure we don't do this (and end-users are informed that things will break). Should we have tests to validate that this is working as expected?

Reply:
good call - added